persistence/gov.go

```diff
@@ -9,893 +14,85 @@ import (
+	return v, err
 }
 
-func (p PostgresContext) GetMessageEditStakeServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageEditStakeServiceNodeFeeOwner)
+func (p PostgresContext) SetParam(paramName string, value any) error {
+	return p.setParamOrFlag(paramName, value, nil)
 }
 
-func (p PostgresContext) GetMessageUnstakeServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageUnstakeServiceNodeFeeOwner)
+func (p PostgresContext) InitFlags() error {
+	// TODO: not implemented
+	return nil
 }
 
-func (p PostgresContext) GetMessagePauseServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessagePauseServiceNodeFeeOwner)
+func (p PostgresContext) GetIntFlag(flagName string, height int64) (value int, enabled bool, err error) {
+	return getParamOrFlag[int](p, schema.FlagsTableName, flagName, height)
 }
 
-func (p PostgresContext) GetMessageUnpauseServiceNodeFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageUnpauseServiceNodeFeeOwner)
+func (p PostgresContext) GetStringFlag(flagName string, height int64) (value string, enabled bool, err error) {
+	return getParamOrFlag[string](p, schema.FlagsTableName, flagName, height)
 }
 
-func (p PostgresContext) GetMessageChangeParameterFeeOwner() ([]byte, error) {
-	return p.GetBytesParam(types.MessageChangeParameterFeeOwner)
+func (p PostgresContext) GetBytesFlag(flagName string, height int64) (value []byte, enabled bool, err error) {
+	return getParamOrFlag[[]byte](p, schema.FlagsTableName, flagName, height)
 }
 
-func (p PostgresContext) GetServiceNodesPerSessionAt(height int64) (int, error) {
-	return p.GetIntParam(types.ServiceNodesPerSessionParamName)
+func (p PostgresContext) SetFlag(flagName string, value any, enabled bool) error {
+	return p.setParamOrFlag(flagName, value, &enabled)
 }
 
-func (p PostgresContext) InitParams() error {
-	ctx, conn, err := p.GetCtxAndConnection()
-	if err != nil {
-		return err
+// setParamOrFlag simply wraps the call to the generic function with the supplied underlying type
+func (p PostgresContext) setParamOrFlag(name string, value any, enabled *bool) error {
+	switch t := value.(type) {
+	case int:
+		return setParamOrFlag(p, name, t, enabled)
+	case int32:
+		return setParamOrFlag(p, name, t, enabled)
+	case int64:
+		return setParamOrFlag(p, name, t, enabled)
+	case []byte:
+		return setParamOrFlag(p, name, t, enabled)
+	case string:
+		return setParamOrFlag(p, name, t, enabled)
+	default:
+		log.Fatalf("unhandled paramType %T for value %v", t, value)
 	}
-	_, err = conn.Exec(ctx, schema.InsertParams(genesis.DefaultParams()))
-	return err
+	return fmt.Errorf("how did you get here?")
 }
 
-// IMPROVE(team): Switch to generics
-func (p PostgresContext) SetParam(paramName string, paramValue interface{}) error {
+// setParamOrFlag sets a param or a flag.
+// If `enabled` is nil, we are dealing with a param, otherwise it's a flag
```

Comment:
**NOT ACTIONABLE**: I thought about this for a moment but ultimately really like the use of the `nil` 💯